# Description: use gpgme_check_version () since gpgme_new () requires it now
# Bug Gentoo: http://bugs.gentoo.org/show_bug.cgi?id=281729
# Bug-Ubuntu: https://bugs.launchpad.net/bugs/514950
# Bug: https://bugzilla.gnome.org/show_bug.cgi?id=607423
Index: seahorse-plugins-2.30.0/plugins/nautilus/seahorse-tool.c
===================================================================
--- seahorse-plugins-2.30.0.orig/plugins/nautilus/seahorse-tool.c	2010-02-13 15:31:34.000000000 +0100
+++ seahorse-plugins-2.30.0/plugins/nautilus/seahorse-tool.c	2010-04-08 02:46:54.000000000 +0200
@@ -170,6 +170,9 @@
         recips = cryptui_prompt_recipients (keyset, _("Choose Recipients"), &signer);
         
         if (recips) {
+            gpgme_check_version (NULL);
+            gerr = gpgme_engine_check_version (GPGME_PROTOCOL_OpenPGP);
+            g_return_val_if_fail (GPG_IS_OK (gerr), NULL);
             
             gerr = gpgme_new (&ctx);
             g_return_val_if_fail (GPG_IS_OK (gerr), NULL);
@@ -328,6 +331,10 @@
         id = cryptui_keyset_key_raw_keyid (keyset, signer);
         g_free (signer);
         
+        gpgme_check_version (NULL);
+        gerr = gpgme_engine_check_version (GPGME_PROTOCOL_OpenPGP);
+        g_return_val_if_fail (GPG_IS_OK (gerr), NULL);
+
         gerr = gpgme_new (&ctx);
         g_return_val_if_fail (GPG_IS_OK (gerr), NULL);
         
Index: seahorse-plugins-2.30.0/agent/seahorse-agent-cache.c
===================================================================
--- seahorse-plugins-2.30.0.orig/agent/seahorse-agent-cache.c	2010-01-25 03:06:50.000000000 +0100
+++ seahorse-plugins-2.30.0/agent/seahorse-agent-cache.c	2010-04-08 02:46:54.000000000 +0200
@@ -237,6 +237,7 @@
         g_cache =
             g_hash_table_new_full (g_str_hash, g_str_equal, NULL, destroy_cache_item);
 
+        gpgme_check_version (NULL);
         err = gpgme_engine_check_version (proto);
         g_return_if_fail (GPG_IS_OK (err));
         
Index: seahorse-plugins-2.30.0/libseahorse/seahorse-pgp-operation.c
===================================================================
--- seahorse-plugins-2.30.0.orig/libseahorse/seahorse-pgp-operation.c	2010-01-25 03:06:50.000000000 +0100
+++ seahorse-plugins-2.30.0/libseahorse/seahorse-pgp-operation.c	2010-04-08 02:46:54.000000000 +0200
@@ -323,6 +323,7 @@
     gpgme_error_t err;
     gpgme_ctx_t ctx;
  
+    gpgme_check_version (NULL);
     err = gpgme_engine_check_version (proto);
     g_return_if_fail (GPG_IS_OK (err));
    
